src/Discord.Net.Core/Entities/Interactions/Modals/ModalBuilder.cs

```diff
@@ -108,21 +110,100 @@ public ModalBuilder AddComponents(List<IMessageComponent> components, int row)
             return this;
         }
 
+        /// <summary>
+        ///     Gets a <see cref="TMessageComponent"/> by the specified <paramref name="customId"/>.
+        /// </summary>
+        /// <typeparam name="TMessageComponent">The type of the component to get.</typeparam>
+        /// <param name="customId">The <see cref="IMessageComponent.CustomId"/> of the component to get.</param>
+        /// <returns>The component of type <typeparamref name="TMessageComponent"/> that was found.</returns>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <typeparamref name="TMessageComponent"/> to be updated was not found.
+        /// </exception>
+        public TMessageComponent GetComponent<TMessageComponent>(string customId) 
+            where TMessageComponent : class, IMessageComponent 
+            => Components.ActionRows?.SelectMany(r => r.Components.OfType<TMessageComponent>()).FirstOrDefault(c => c?.CustomId == customId)
+                ?? throw new ArgumentException($"There is no component of type {typeof(TMessageComponent).Name} with the specified custom ID in this modal builder.", nameof(customId));
+
+        /// <summary>
+        ///     Updates a <see cref="TextInputComponent"/> by the specified <paramref name="customId"/>.
+        /// </summary>
+        /// <param name="customId">The <see cref="TextInputComponent.CustomId"/> of the input to update.</param>
+        /// <param name="updateTextInput">An action that configures the updated text input.</param>
+        /// <returns>The current builder.</returns>
+        public ModalBuilder UpdateTextInput(string customId, Action<TextInputBuilder> updateTextInput)
+        {
+            var component = GetComponent<TextInputComponent>(customId);
+            var row = Components.ActionRows.First(r => r.Components.Contains(component));
+
+            var builder = new TextInputBuilder
+            {
+                Label = component.Label,
+                CustomId = component.CustomId,
+                Style = component.Style,
+                Placeholder = component.Placeholder,
+                MinLength = component.MinLength,
+                MaxLength = component.MaxLength,
+                Required = component.Required,
+                Value = component.Value
+            };
+
+            updateTextInput(builder);
```

Comment:
I don't see a method called `updateTextInput` 🤔
typo?

Reply:
it's an `Action<TextInputBuilder>` parameter of this method

Reply:
Ah, yup, missed that.